src/lib.rs

```diff
@@ -351,3 +460,59 @@ fn convert_response_to_query_response(res: skar_client::QueryResponse) -> Result
         },
     })
 }
+
+/// Construct response and centralize error mapping for calling function.
+fn compose_pyarrow_response(
+    archive_height: Option<u64>,
+    next_block: u64,
+    total_execution_time: u64,
+    blocks: PyObject,
+    transactions: PyObject,
+    logs: PyObject,
+) -> Result<QueryResponseArrow> {
+    Ok(QueryResponseArrow {
+        archive_height: archive_height
+            .map(|h| h.try_into())
+            .transpose()
+            .context("convert height")?,
+        next_block: next_block.try_into().context("convert next_block")?,
+        total_execution_time: total_execution_time
+            .try_into()
+            .context("convert total_execution_time")?,
+        data: QueryArrowResponse {
+            blocks,
+            transactions,
+            logs,
+        },
+    })
+}
+
+/// Uses RecordBatchReader to convert vector of ArrayBatch to reader by c-interface
+/// and then crates table from this reader with method from_batches
+fn convert_batch_to_pyarrow_table<'py>(
+    py: Python<'py>,
+    pyarrow: &'py PyModule,
+    batches: Vec<ArrowBatch>,
+) -> PyResult<PyObject> {
+    let schema = batches.first().unwrap().schema.fields.clone();
```

Comment:
Batches can be empty

Reply:
What behavior do we want in this case?

Reply:
For now make it to return python None in case batch is empty